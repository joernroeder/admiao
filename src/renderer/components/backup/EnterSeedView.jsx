import React from 'react'
import { Router } from '@reach/router'

import { ThemeProvider } from 'emotion-theming'
import { darkTheme } from '../../utils/theme'

import GridWrap from '../GridWrap'
import Cell from '../Cell'

import Intro from './enterseed/Intro'
import IntroSidebar from './enterseed/IntroSidebar'

import Enter from './enterseed/Enter'
import EnterSidebar from './enterseed/EnterSidebar'

const EnterSeed = () => {
	return (
		<ThemeProvider theme={darkTheme}>
			<GridWrap minHeight={'100vh'}>
				<Cell gridOffset={2} gridColumn={3} mt={9}>
					<Router>
						<IntroSidebar default />
						<EnterSidebar path="enter" />
					</Router>
				</Cell>
				<Cell gridOffset={1} gridColumn={5}>
					<Router>
						<Intro default />
						<Enter path="enter" />
					</Router>
				</Cell>
			</GridWrap>
		</ThemeProvider>
	)
}

export default EnterSeed
