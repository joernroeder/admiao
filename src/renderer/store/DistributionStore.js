import React, { createContext, useContext, useReducer, useEffect } from 'react'

const localStorageKey = 'DistributionState'

const initialState = {
	distributionType: undefined,
	distributionIdentifier: undefined,
}

const DistributionTypes = {
	LOCAL_PRINT: 'LOCAL_PRINT',
	REMOTE_PRINT: 'REMOTE_PRINT',
	DIGITAL: 'DIGITAL',
	FILE_SYSTEM: 'FILE_SYSTEM',
}

function reducer(state, action) {
	// sets Distribution Type
	if (Object.keys(DistributionTypes).includes(action.type)) {
		return {
			...state,
			distributionType: DistributionTypes[action.type],
			distributionIdentifier: action.payload || undefined,
		}
	}

	switch (action.type) {
		case 'RESET': {
			localStorage.removeItem(localStorageKey)

			return {
				...initialState,
			}
		}

		case 'SET_IDENTIFIER': {
			return {
				...state,
				distributionIdentifier: action.payload,
			}
		}

		default:
			return state
	}
}

const DistributionStateContext = createContext(undefined)
const DistributionDispatchContext = createContext(undefined)

function DistributionProvider({ children }) {
	const [state, dispatch] = useReducer(
		reducer,
		initialState,
		initialState => {
			const previousState = localStorage.getItem(localStorageKey)

			return previousState ? JSON.parse(previousState) : initialState
		}
	)

	// store distributionType in local storage.
	// This stores the distribution type to improve the backup process.
	useEffect(() => {
		if (state.isInitialState) {
			return
		}

		// filter out empty and initial states
		if (!Object.values(state).filter(Boolean).length) {
			return
		}

		console.log(Object.values(state).filter(Boolean).length)
		localStorage.setItem(localStorageKey, JSON.stringify(state))
	}, [state])

	return (
		<DistributionStateContext.Provider value={state}>
			<DistributionDispatchContext.Provider value={dispatch}>
				{children}
			</DistributionDispatchContext.Provider>
		</DistributionStateContext.Provider>
	)
}

function useDistributionState() {
	const context = useContext(DistributionStateContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionState must be used within a DistributionProvider'
		)
	}
	return context
}

function useDistributionDispatch() {
	const context = useContext(DistributionDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionDispatch must be used within a DistributionProvider'
		)
	}
	return context
}

export {
	DistributionProvider,
	useDistributionState,
	useDistributionDispatch,
	DistributionTypes,
}
