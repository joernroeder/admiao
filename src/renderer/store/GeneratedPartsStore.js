import React, { createContext, useContext, useReducer } from 'react'

const initialState = {
	parts: [],
	confirmed: [],
}

function reducer(state, action) {
	switch (action.type) {
		case 'SET_PARTS': {
			console.log('setting parts in store', action)
			return {
				...state,
				confirmed: Array(action.payload.length).fill(false),
				parts: action.payload,
			}
		}

		case 'CONFIRMED': {
			const updatedConfirmed = {
				confirmed: state.confirmed.map((isConfirmed, index) => {
					if (index === action.payload) {
						return true
					}

					return isConfirmed
				}),
			}

			return {
				...state,
				...updatedConfirmed,
			}
		}

		case 'RESET': {
			return {
				...initialState,
			}
		}

		default:
			return state
	}
}

const GeneratedPartsStateContext = createContext(undefined)
const GeneratedPartsDispatchContext = createContext(undefined)

function GeneratedPartsProvider({ children }) {
	const [state, dispatch] = useReducer(reducer, initialState)

	return (
		<GeneratedPartsStateContext.Provider value={state}>
			<GeneratedPartsDispatchContext.Provider value={dispatch}>
				{children}
			</GeneratedPartsDispatchContext.Provider>
		</GeneratedPartsStateContext.Provider>
	)
}

function useGeneratedParts() {
	const context = useContext(GeneratedPartsStateContext)

	if (context === undefined) {
		throw new Error(
			'useGeneratedParts must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

function useGeneratedPartsDispatch() {
	const context = useContext(GeneratedPartsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useDistributionDispatch must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

export { GeneratedPartsProvider, useGeneratedParts, useGeneratedPartsDispatch }
