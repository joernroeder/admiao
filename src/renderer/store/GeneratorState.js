import React, { createContext, useContext, useState } from 'react'

const STATES = {
	loading: 'LOADING',
	successfullyGenerated: 'SUCCESSFULLY_GENERATED',
	generationFailed: 'GENERATION_FAILED',
	successfullyDistributed: 'SUCCESSFULLY_DISTRIBUTED',
	distributionFailed: 'DISTRIBUTION_FAILED',
}

const GeneratorStateContext = createContext(undefined)

function GeneratorStateProvider({ children }) {
	const [generatorState, setGeneratorState] = useState({
		state: undefined,
		pendingIdentifier: null,
		error: null,
		// additional data returned by the ipc methods may get stored here.
		// todo be more precise/refactor once all distribution types are implemented
		meta: {},
	})

	return (
		<GeneratorStateContext.Provider
			value={[generatorState, setGeneratorState]}
		>
			{children}
		</GeneratorStateContext.Provider>
	)
}

function useGeneratorState() {
	const context = useContext(GeneratorStateContext)

	if (context === undefined) {
		throw new Error(
			'useGeneratedParts must be used within a GeneratedPartsProvider'
		)
	}
	return context
}

export { GeneratorStateProvider, useGeneratorState, STATES }
