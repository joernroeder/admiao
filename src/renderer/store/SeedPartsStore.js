import React, { createContext, useContext, useEffect, useReducer } from 'react'

const localStorageKey = 'SeedPartsState'

const initialState = {
	uniquePartsN: 3,
	requiredPartsT: 2,

	partsPerDisk: 1,
	requiredDisks: 3,

	recoveredFromPreviousSelection: false,
	isInitialState: true,
}

const calcDisks = (uniquePartsN, requiredPartsT) => {
	const partsPerDisk = requiredPartsT - 1
	const requiredDisks = Math.ceil(uniquePartsN / partsPerDisk)

	return {
		partsPerDisk,
		requiredDisks,
	}
}

function reducer(state, action) {
	const { uniquePartsN, requiredPartsT } = state

	const isModifiedState = {
		recoveredFromPreviousSelection: false,
		isInitialState: false,
	}

	switch (action.type) {
		case 'INCREMENT_REQUIRED': {
			const updatedRequiredPartsT = Math.min(
				requiredPartsT + 1,
				uniquePartsN
			)

			return {
				...state,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(state.uniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'DECREMENT_REQUIRED': {
			const updatedRequiredPartsT = Math.max(requiredPartsT - 1, 1)

			return {
				...state,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(state.uniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'INCREMENT_UNIQUE': {
			const updatedUniquePartsN = uniquePartsN + 1

			return {
				...state,
				uniquePartsN: updatedUniquePartsN,
				...calcDisks(updatedUniquePartsN, state.requiredPartsT),
				...isModifiedState,
			}
		}

		case 'DECREMENT_UNIQUE': {
			const updatedUniquePartsN = Math.max(uniquePartsN - 1, 1)
			// also decrement required parts if they would exceed otherwise
			const updatedRequiredPartsT = Math.min(
				updatedUniquePartsN,
				requiredPartsT
			)

			return {
				...state,
				uniquePartsN: updatedUniquePartsN,
				requiredPartsT: updatedRequiredPartsT,
				...calcDisks(updatedUniquePartsN, updatedRequiredPartsT),
				...isModifiedState,
			}
		}

		case 'CONFIRM': {
			return {
				...state,
				...isModifiedState,
			}
		}

		case 'RESET': {
			localStorage.removeItem(localStorageKey)

			return {
				...initialState,
			}
		}

		default:
			return state
	}
}

const SeedPartsStateContext = createContext(undefined)
const SeedPartsDispatchContext = createContext(undefined)

function SeedPartsProvider({ children }) {
	// uses a previous state from local storage if available or uses the initial one defined above.
	const [state, dispatch] = useReducer(
		reducer,
		initialState,
		initialState => {
			const previousState = localStorage.getItem(localStorageKey)

			return previousState
				? {
						...JSON.parse(previousState),
						recoveredFromPreviousSelection: true,
				  }
				: initialState
		}
	)

	// store current state in local storage.
	// This is used to improve the backup process if the user paused to print the
	// seed templates.
	useEffect(() => {
		if (state.isInitialState) {
			return
		}

		localStorage.setItem(localStorageKey, JSON.stringify(state))
	}, [state])

	return (
		<SeedPartsStateContext.Provider value={state}>
			<SeedPartsDispatchContext.Provider value={dispatch}>
				{children}
			</SeedPartsDispatchContext.Provider>
		</SeedPartsStateContext.Provider>
	)
}

function useSeedPartsState() {
	const context = useContext(SeedPartsStateContext)

	if (context === undefined) {
		throw new Error(
			'useSeedPartsState must be used within a SeedPartsProvider'
		)
	}
	return context
}

function useSeedPartsDispatch() {
	const context = useContext(SeedPartsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useSeedPartsDispatch must be used within a SeedPartsProvider'
		)
	}
	return context
}

export { SeedPartsProvider, useSeedPartsState, useSeedPartsDispatch }
