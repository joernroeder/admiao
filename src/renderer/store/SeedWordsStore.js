import React, { createContext, useContext, useReducer } from 'react'
const bip39 = require('bip39')

const initialWords = [
	/*
	'answer',
	'dice',
	'warm',
	'cup',
	'shock',
	'orbit',
	'pistol',
	'seat',
	'modify',
	'matrix',
	'genre',
	'renew',
	*/
]

const initialState = {
	words: initialWords,
	total: 12, // todo handle other seeds 18, 24
	isConfirmed: false,
	isValid: getIsValid(initialWords),
}

function getIsValid(words) {
	return bip39.validateMnemonic(words.join(' '))
}

function wordsReducer(state, action) {
	const { words } = state

	switch (action.type) {
		case 'APPEND': {
			const updatedWords = [...words, action.payload]

			return {
				...state,
				words: updatedWords,
				isValid: getIsValid(updatedWords),
			}
		}

		case 'RESET': {
			return {
				...state,
				words: [],
				isConfirmed: false,
				isValid: false,
			}
		}

		case 'CLEANUP_SEED': {
			return {
				...state,
				words: [],
			}
		}

		case 'POP': {
			const updatedWords = [...words.slice(0, -1)]

			return {
				...state,
				words: updatedWords,
				isValid: getIsValid(updatedWords),
			}
		}

		case 'CONFIRM': {
			return {
				...state,
				isConfirmed: true,
			}
		}

		default:
			return state
	}
}

const SeedWordsStateContext = createContext(undefined)
const SeedWordsDispatchContext = createContext(undefined)

function SeedWordsProvider({ children }) {
	const [state, dispatch] = useReducer(wordsReducer, initialState)

	return (
		<SeedWordsStateContext.Provider value={state}>
			<SeedWordsDispatchContext.Provider value={dispatch}>
				{children}
			</SeedWordsDispatchContext.Provider>
		</SeedWordsStateContext.Provider>
	)
}

function useSeedWordsState() {
	const context = useContext(SeedWordsStateContext)

	if (context === undefined) {
		throw new Error(
			'useSeedWordsState must be used within a SeedWordsProvider'
		)
	}
	return context
}

function useSeedWordsDispatch() {
	const context = useContext(SeedWordsDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useSeedWordsDispatch must be used within a SeedWordsProvider'
		)
	}
	return context
}

export {
	SeedWordsProvider,
	useSeedWordsState,
	useSeedWordsDispatch,
	wordsReducer,
}
