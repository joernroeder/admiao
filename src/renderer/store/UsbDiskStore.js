import React, { createContext, useContext, useReducer } from 'react'

const initialState = {
	useManagedDisks: true,
}

function reducer(state, action) {
	switch (action.type) {
		case 'USE_PREPARED_DISKS': {
			return {
				...state,
				useManagedDisks: true,
			}
		}

		case 'DISABLE_DISK_MANAGEMENT': {
			return {
				...state,
				useManagedDisks: false,
			}
		}

		default:
			return state
	}
}

const UsbDiskStateContext = createContext(undefined)
const UsbDiskDispatchContext = createContext(undefined)

function UsbDiskProvider({ children }) {
	const [state, dispatch] = useReducer(reducer, initialState)

	return (
		<UsbDiskStateContext.Provider value={state}>
			<UsbDiskDispatchContext.Provider value={dispatch}>
				{children}
			</UsbDiskDispatchContext.Provider>
		</UsbDiskStateContext.Provider>
	)
}

function useUsbDiskState() {
	const context = useContext(UsbDiskStateContext)

	if (context === undefined) {
		throw new Error('useUsbDiskState must be used within a UsbDiskProvider')
	}
	return context
}

function useUsbDiskDispatch() {
	const context = useContext(UsbDiskDispatchContext)

	if (context === undefined) {
		throw new Error(
			'useUsbDiskDispatch must be used within a UsbDiskProvider'
		)
	}
	return context
}

export { UsbDiskProvider, useUsbDiskState, useUsbDiskDispatch }
