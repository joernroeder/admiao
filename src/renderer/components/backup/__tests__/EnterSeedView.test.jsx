import React from 'react'

import { act, fireEvent, render } from '@testing-library/react'
import { DistributionProvider } from '../../../store/DistributionStore'

import EnterSeedView from '../EnterSeedView'

describe('Enter Seed', () => {
	beforeEach(() => {
		global.ipcRenderer = {
			send: jest.fn(),
			once: jest.fn(),
			on: jest.fn(),
			removeListener: jest.fn(),
		}
	})

	const renderWithProviders = (children, renderFunction = render) => {
		return renderFunction(
			<DistributionProvider>{children}</DistributionProvider>
		)
	}

	test('it should render correctly', () => {
		const { getByText, container } = renderWithProviders(<EnterSeedView />)

		expect(getByText('Enter Seed')).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
