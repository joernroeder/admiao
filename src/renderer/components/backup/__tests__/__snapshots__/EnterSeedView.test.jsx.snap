// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enter Seed it should render correctly 1`] = `
.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  min-height: 100vh;
  margin-left: -8px;
  margin-right: -8px;
  background-color: rgba(0,0,0,1);
  color: rgba(255,255,255,1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 100vh;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 72px;
  padding-left: 8px;
  padding-right: 8px;
  margin-left: 14.285714285714285%;
  width: 21.428571428571427%;
}

.emotion-0 {
  color: rgba(255,255,255,1);
  margin-top: 0;
  margin-bottom: 0;
  font-size: 1.3125rem;
  line-height: 2rem;
  -webkit-transform: translateY(0.5078125rem);
  -ms-transform: translateY(0.5078125rem);
  transform: translateY(0.5078125rem);
  font-family: 'SpaceGrotesk',sans-serif;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 4.5rem;
    line-height: 4.5rem;
    -webkit-transform: translateY(0.5625rem);
    -ms-transform: translateY(0.5625rem);
    transform: translateY(0.5625rem);
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 4.5rem;
    line-height: 4.5rem;
    -webkit-transform: translateY(0.5625rem);
    -ms-transform: translateY(0.5625rem);
    transform: translateY(0.5625rem);
  }
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
  padding-right: 8px;
  margin-left: 7.142857142857142%;
  width: 35.714285714285715%;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-top: 168px;
}

.emotion-2 {
  color: rgba(255,255,255,1);
  margin-bottom: 16px;
  margin-top: 0;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  -webkit-transform: translateY(0.421875rem);
  -ms-transform: translateY(0.421875rem);
  transform: translateY(0.421875rem);
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
}

.emotion-3 {
  color: rgba(0,0,0,1);
  background-color: rgba(255,255,255,1);
  margin-bottom: 32px;
  margin-top: 0;
  padding: 8px;
  padding-left: 16px;
  margin-right: -16px;
  font-size: 0.875rem;
  line-height: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.5rem;
  -webkit-transform: translateY(0.421875rem);
  -ms-transform: translateY(0.421875rem);
  transform: translateY(0.421875rem);
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  font-variant-ligatures: common-ligatures;
  color: rgba(0,0,0,1);
  background-color: rgba(255,255,255,1);
}

.emotion-5 {
  border-width: 2px;
  border-color: rgba(255,255,255,1);
  border-style: solid;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-size: 0.875rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
  font-size: 0.875rem;
  line-height: 1.5rem;
  -webkit-transform: translateY(0.421875rem);
  -ms-transform: translateY(0.421875rem);
  transform: translateY(0.421875rem);
  color: rgba(255,255,255,1);
  background-color: transparent;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  box-sizing: border-box;
  position: relative;
  z-index: 10;
  overflow: hidden;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-rendering: geometricPrecision;
  -webkit-font-smoothing: antialiased;
  cursor: pointer;
}

.emotion-5 .icon {
  display: inline-block;
  -webkit-transition: -webkit-transform 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in;
  transition: transform 0.25s ease-in;
}

.emotion-5:hover .icon {
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.emotion-5:hover .icon.next {
  -webkit-transform: translateX(0.4em);
  -ms-transform: translateX(0.4em);
  transform: translateX(0.4em);
}

.emotion-5:hover .icon.prev {
  -webkit-transform: translateX(-0.4em);
  -ms-transform: translateX(-0.4em);
  transform: translateX(-0.4em);
}

.emotion-5:hover .icon.flip {
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.emotion-5 > span:first-of-type {
  position: relative;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-4 {
  background-color: rgba(255,255,255,0.15);
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  -webkit-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -ms-transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  transform: skew( -14deg, 0 ) translateX( -110% ) translateY( 0% );
  -webkit-transition: -webkit-transform 0.25s ease-in,opacity 0.25s ease-in;
  -webkit-transition: transform 0.25s ease-in,opacity 0.25s ease-in;
  transition: transform 0.25s ease-in,opacity 0.25s ease-in;
}

*:hover > .emotion-4,
*:focus > .emotion-4 {
  opacity: 1;
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  -ms-transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
  transform: translateX( -1.9em ) translateY(0) skew( -14deg, 0 );
}

<div
  class="emotion-8"
>
  <div
    class="emotion-1"
  >
    <div
      role="group"
      style="outline: none;"
      tabindex="-1"
    >
      <h2
        class="emotion-0"
        color="black"
        font-size="1,2,2"
      >
        Enter Seed
      </h2>
    </div>
  </div>
  <div
    class="emotion-7"
  >
    <div
      role="group"
      style="outline: none;"
      tabindex="-1"
    >
      <div
        class="emotion-6"
      >
        <p
          class="emotion-2"
          color="black"
          font-size="0"
        >
          In the following section you will asked to enter the words of your seed one by one. Make sure no human or robot looks over your shoulder or at your screen.
        </p>
        <p
          class="emotion-3"
          color="white"
          font-size="0"
        >
          <strong>
            Remember:
          </strong>
           This should be the only time you
           
          <u>
            ever ever
          </u>
           type those words into a computer apart from restoring purposes.
        </p>
        <a
          class="emotion-5"
          href="//enter"
        >
          <span>
            Yes, i am alone.
             
            <span
              class="icon next"
            >
              →
            </span>
          </span>
          <span
            class="emotion-4"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
